File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.cpp
class Solution {
public:
    
    bool isVowel(char c) {
        char low = tolower(c);
        return low == 'a' || low == 'e' || low == 'i' || low == 'o' || low == 'u';
    }
    string reverseVowels(string s) {
        //use two pointer to search. 
        int lo = 0;
        int hi = s.length()-1;
        while(lo < hi){
            while (lo < hi && !isVowel(s[lo])) {
                lo++;
            }
            while (lo < hi && !isVowel(s[hi])) {
                hi--;
            }
            if (lo < hi) {
                swap(s[lo], s[hi]);
                lo++;
                hi--;
            }

        }
        return s;
    }
    
};

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.py
class Solution(object):
    def reverseVowels(self, s):
        s = list(s)
        string = 'ueoaiUEOAI'
        lo = 0
        hi = len(s) - 1
        while(lo < hi):
            while(lo <= hi and string.find(s[lo]) == -1):
                lo += 1
            while(lo <= hi and string.find(s[hi]) == -1):
                hi -= 1
            if(lo < hi):
                s[lo],s[hi] = s[hi],s[lo]
                lo += 1
                hi -= 1
        return "".join(s)
        